albumentations/augmentations/transforms.py

```diff
@@ -3107,3 +3108,45 @@ def apply(self, img, transforms=(), **params):
 
     def get_transform_init_args_names(self):
         return ("brightness", "contrast", "saturation", "hue")
+
+
+class Sharpen(ImageOnlyTransform):
+    """Sharpen the input image and overlays the result with the original image.
+
+    Args:
+        alpha ((float, float)): range to choose the visibility of the sharpened image. At 0, only the original image is
+            visible, at 1.0 only its sharpened version is visible. Default: (0.2, 0.5).
+        lightness ((float, float)): range to choose the lightness of the sharpened image. Default: (0.5, 1.0).
+        p (float): probability of applying the transform. Default: 0.5.
+
+    Targets:
+        image
+    """
+
+    def __init__(self, alpha=(0.2, 0.5), lightness=(0.5, 1.0), always_apply=False, p=0.5):
+        super(Sharpen, self).__init__(always_apply, p)
+        self.alpha = self.__check_values(to_tuple(alpha, 0.0), name="alpha", bounds=(0.0, 1.0))
+        self.lightness = self.__check_values(to_tuple(lightness, 0.0), name="lightness")
+
+    @staticmethod
+    def __check_values(value, name, bounds=(0, float("inf"))):
+        if not bounds[0] <= value[0] <= value[1] <= bounds[1]:
+            raise ValueError("{} values should be between {}".format(name, bounds))
+        return value
+
+    def get_params(self):
+        return {"alpha": random.uniform(*self.alpha), "lightness": random.uniform(*self.lightness)}
+
+    def _generate_sharpening_matrix(self, alpha_sample, lightness_sample):
+        matrix_nochange = np.array([[0, 0, 0], [0, 1, 0], [0, 0, 0]], dtype=np.float32)
+        matrix_effect = np.array([[-1, -1, -1], [-1, 8 + lightness_sample, -1], [-1, -1, -1]], dtype=np.float32)
+
+        matrix = (1 - alpha_sample) * matrix_nochange + alpha_sample * matrix_effect
+        return matrix
+
+    def apply(self, img, alpha=0.0, lightness=0.0, **params):
+        sharpening_matrix = self._generate_sharpening_matrix(alpha_sample=alpha, lightness_sample=lightness)
```

Comment:
Might be better to create a matrix in `get_params`?

Reply:
For sake of speed in case of multi-image & replay processing - computing matrix in `get_params` would be faster. That was my suggestion to compute _generate_sharpening_matrix in `apply` call. @KiriLev would you please move it to `get_params`. The rest looks good to me.

Reply:
moved!